File: Arrays/One Dimensional Arrays/MaxandMin.cpp
// Method 1

#include <bits/stdc++.h>
using namespace std;

void func(int arr[], int n){
    int largest, smallest;

    largest = INT_MIN;
    smallest = INT_MAX;

    for (int i = 0; i < n; i ++){
        
        // Using if condition to check
        if (arr[i] > largest){
            largest = arr[i];
        }

        if (arr[i] < smallest){
            smallest = arr[i];
        }


    }

    cout << "largest: " << largest << endl;
    cout << "smallest: " << smallest << endl;

}

int main(){
    
    int n, key;
    cin >> n;

    int arr[1000];
    for (int i = 0; i < n; i++){
        cin >> arr[i];
    }

    func(arr, n);
    return 0;
}

// method 2

#include <bits/stdc++.h>
using namespace std;

void func(int arr[], int n){
    int largest, smallest;

    largest = INT_MIN;
    smallest = INT_MAX;

    for (int i = 0; i < n; i ++){
        
        // using inbuilt maths
        largest = max(largest, arr[i]);
        smallest = min(smallest, arr[i]);

    }

    cout << "largest: " << largest << endl;
    cout << "smallest: " << smallest << endl;

}

int main(){
    
    int n, key;
    cin >> n;

    int arr[1000];
    for (int i = 0; i < n; i++){
        cin >> arr[i];
    }

    func(arr, n);
    return 0;
}

File: Arrays/One Dimensional Arrays/MaxandMin.py
# using standards

arr = [-1, 2, 3, 4, 5]

def function(arr):
    
   
    
    largest = arr[0]
    smallest = arr[0]
    for i in range(1, len(arr)):
        if (arr[i] > largest):
            largest = arr[i]
        if (arr[i] < smallest):
            smallest = arr[i]
            
    print(largest, smallest)
    

function(arr)

# using language tools

arr = [-1, 2, 3, 4, 5]

def function(arr):
    
    largest = float('-inf')
    smallest = float('inf')
    
    
    for i in range(0, len(arr)):
        if (arr[i] > largest):
            largest = arr[i]
        if (arr[i] < smallest):
            smallest = arr[i]
            
    print(largest, smallest)
    
function(arr)
